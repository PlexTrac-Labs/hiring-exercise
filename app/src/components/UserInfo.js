import React from "react";

export default function UserInfo({ user }) {
  return (
    <div className="Profile">
      <div className="User-firstName" class="mb-3">
        {user.firstName}
      </div>
      <div className="User-lastName">{user.lastName}</div>
      <div className="User-userName">{user.username}</div>
      <div className="User-email">{user.email}</div>
      <div className="User-birthYear">{user.birthYear}</div>
      <div className="User-favColor">{user.favColor}</div>
    </div>
  );
}
