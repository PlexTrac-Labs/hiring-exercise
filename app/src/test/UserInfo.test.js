import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import UserInfo from "../components/UserInfo";

//Create user to use for testing
const user = {
  firstName: "John",
  lastName: "Doe",
  username: "jdoe",
  email: "johndoe@mail.com",
  birthYear: "1999",
  favColor: "Blue"
};

//Test to make sure the App, entry point to everything works.
it("renders correctly", async () => {
  const tree = renderer.create(<UserInfo user={user} />).toJSON();
  expect(tree).toMatchSnapshot();
});
