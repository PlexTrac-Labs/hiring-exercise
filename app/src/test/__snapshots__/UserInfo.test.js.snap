// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Profile"
>
  <h2
    className="User-firstName mb-3"
  >
    John
     
    Doe
  </h2>
  <dl>
    <dt>
      User Name
    </dt>
    <dd
      className="User-userName"
    >
      jdoe
    </dd>
    <dt>
      Email
    </dt>
    <dd
      className="User-email"
    >
      johndoe@mail.com
    </dd>
    <dt>
      Year of Birth
    </dt>
    <dd
      className="User-birthYear"
    >
      1999
    </dd>
    <dt>
      Favorite Color
    </dt>
    <dd
      className="User-favColor"
    >
      Blue
    </dd>
  </dl>
</div>
`;
